server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -138,17 +139,21 @@ public static TaskManager createTaskManagerWithClusterSettings(
         ThreadPool threadPool,
         Set<String> taskHeaders
     ) {
-        final TaskManager taskManager = new TaskManager(settings, threadPool, taskHeaders);
+        final TaskManager taskManager = new TaskManager(settings, threadPool, taskHeaders, clusterSettings);
         clusterSettings.addSettingsUpdateConsumer(TASK_RESOURCE_CONSUMERS_ENABLED, taskManager::setTaskResourceConsumersEnabled);
         return taskManager;
     }
 
     public TaskManager(Settings settings, ThreadPool threadPool, Set<String> taskHeaders) {
+        this(settings, threadPool, taskHeaders, null);
+    }
```

Comment:
I believe this constructor is preserved to keep the existing tests unmodified. If the effort is small, can we refactor them and remove this constructor completely?

Reply:
There are 10+ tests which are using this constructor, passing null argument for each of these constructor usages may create confusion. Thoughts?